org.eclipse.wildwebdeveloper.xml/src/org/eclipse/wildwebdeveloper/xml/XMLExtensionRegistry.java

```diff
@@ -0,0 +1,63 @@
+package org.eclipse.wildwebdeveloper.xml;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.eclipse.core.runtime.IConfigurationElement;
+import org.eclipse.core.runtime.IRegistryChangeEvent;
+import org.eclipse.core.runtime.IRegistryChangeListener;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.Status;
+
+public class XMLExtensionRegistry {
+
+	private static final String EXTENSION_POINT_ID = Activator.PLUGIN_ID + ".xmllsExtension"; //$NON-NLS-1$
+	private Map<IConfigurationElement, String> extensions = new HashMap<>();
+	private boolean outOfSync = true;
+
+	public XMLExtensionRegistry() {
+			Platform.getExtensionRegistry().addRegistryChangeListener(new IRegistryChangeListener() {
+			@Override
+			public void registryChanged(IRegistryChangeEvent event) {
+				outOfSync = true;
+			}
+		}, EXTENSION_POINT_ID);
+	}
+
+	public List<String> getXMLExtensionJars() {
```

Comment:
Aren't those string "path" relative to contributing bundle root?
I think this would need to be clarified in javadoc.